src/vs/workbench/contrib/remote/electron-sandbox/remote.contribution.ts

```diff
@@ -244,3 +250,39 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 	metadata: { description: SaveLocalFileCommand.LABEL, args: [] },
 	handler: SaveLocalFileCommand.handler()
 });
+
+registerAction2(class extends Action2 {
+	constructor() {
+		super({
+			id: 'workbench.remote.action.closeUnusedPorts',
+			title: localize('remote.actions.closeUnusedPorts', 'Close Unused Forwarded Ports'),
+			category: localize('remote.category', 'Remote'),
+			menu: [{
+				id: MenuId.CommandPalette
+			}],
+			precondition: ContextKeyExpr.and(ContextKeyExpr.notEquals(`config.${PORT_AUTO_SOURCE_SETTING}`, PORT_AUTO_SOURCE_SETTING_OUTPUT), RemoteNameContext)
```

Comment:
The contexts you were using @nojaf weren't using the OS of the remote, they were using the local OS. I've looked around but we don't seem to already have broadly useful remote OS context. I removed them, and added a context for whether we're connected to a remote.

I've inlined your strings and also used constants where we have them.

Reply:
Thanks for these changes!